baremetal_network_provisioning/db/bm_nw_provision_db.py

```diff
@@ -349,11 +349,14 @@ def get_bnp_switch_port_map_by_switchid(context, switchid):
 
 def delete_bnp_phys_switch(context, switch_id):
     """Delete physical switch that matches switch_id."""
-    session = context.session
-    with session.begin(subtransactions=True):
-        if switch_id:
-            session.query(models.BNPPhysicalSwitch).filter_by(
-                id=switch_id).delete()
+    try:
+        session = context.session
+        with session.begin(subtransactions=True):
+            if switch_id:
+                session.query(models.BNPPhysicalSwitch).filter_by(
+                    id=switch_id).delete()
+    except exc.NoResultFound:
+        LOG.error('no switch found for switch id: %s', switch_id)
 
```

Comment:
Change the update_bnp_phys_switch_snmpv3 and update_bnp_phys_switch_snmpv2 methods also. Currently these methods doesn't support the change the access_protocols.